### PR TITLE
Decanter 6: NPM Package Updates.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Review this paydown of years of technical debt
- The goal is to be up-to-date, but not to break anything (too much)
- New developer environment
- New up-to-date webpack plugins
- Bourbon 6

# Needed By (Date)
- September 15

# Urgency
- Minor

# Steps to Test

1. Pull down this branch
2. Run `nvm use`
3. Run `npm install`
4. Run the different commands found in the scripts section in `package.json`

### Integration tests

1. Pull this branch into your module/theme project
2. See if it still compiles and/or what updates you have to make. You should be able to take most of what is in the webpack.config into your module/theme.

### Developer environment tests

1. Pull this branch
2. Run the install (nvm use + npm install)
3. Run developer `npm run dev`
4. See if you can find and change each of the 4 developer files (Javascript, Sass, twig, data)
5. Hot reload should be working. Let me know if it isn't
6. Shut down server (CTRL + C)
7. Start it up again `npm run dev` and ensure your changes weren't overwritten

### Preview build test

1. Navigate to the [preview build ](https://deploy-preview-922--decanter-v6.netlify.app/)
2. Perform some linky clicky
3. Compare against current live site or pop over to [preview-site-pr](https://github.com/SU-SWS/decanter-web/pull/142) and follow those steps.

### CODE REVIEW

File: developer/index.js [hunk 1]
```diff
@@ -1 +1,3 @@
 import "./test.js";
```
Comment (sherakama) [inline developer/index.js:1]: Still working on this one.

File: developer/index.twig [hunk 2]
```diff
@@ -10,6 +10,7 @@
     {%- include "@decanter/components/masthead/masthead.twig" -%}
     <main>
       {%- include "./test.twig" -%}
+      <p>Something</[>]
```
Comment (sherakama) [inline developer/index.twig:13]: Still working on this one.

File: developer/webpack.dev.config.js [hunk 3]
```diff
@@ -4,17 +4,12 @@
...
-const autoprefixer =               require('autoprefixer')({ grid: true });
-const FileManagerPlugin =          require('filemanager-webpack-plugin');
-const UglifyJsPlugin =             require("uglifyjs-webpack-plugin");
-const MiniCssExtractPlugin =       require("mini-css-extract-plugin");
-const OptimizeCSSAssetsPlugin =    require("optimize-css-assets-webpack-plugin");
-const WebpackAssetsManifest =      require("webpack-assets-manifest");
-const ExtraWatchWebpackPlugin =    require("extra-watch-webpack-plugin");
-const FixStyleOnlyEntriesPlugin =  require("webpack-fix-style-only-entries");
-const HtmlWebpackPlugin =          require('html-webpack-plugin');
+const path = require('path');
```
Comment (sherakama) [inline developer/webpack.dev.config.js:7]: Still working on this one.

**Comment by yvonnetangsu** [comment]
Wow big update @sherakama 👀 Thank you for doing this!

**Comment by sherakama** [comment]
> Wow big update

340 files with 
`- @charset "UTF-8";`

File: .nvmrc [hunk 4]
```diff
@@ -1 +1 @@
-14
\ No newline at end of file
+18
```
Comment (sherakama) [inline .nvmrc:1]: I suppose I could go to 20 but this seemed good enough.

File: core/src/templates/components/card/card.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "card_media_wrapper_modifier_class": "su-aspect-ratio",
-  "card_media_image_src": "https://placeimg.com/2000/1333/any",
```
Comment (sherakama) [inline core/src/templates/components/card/card.json:3]: placeimg has retired.

File: package.json [hunk 6]
```diff
@@ -8,11 +8,10 @@
   "scripts": {
     "build": "webpack --mode=development",
     "publish": "webpack --mode=production",
-    "watch": "webpack --mode=development --watch --config developer/webpack.dev.config.js",
```
Comment (sherakama) [inline package.json:11]: Watch is built in now.

File: package.json [hunk 7]
```diff
@@ -8,11 +8,10 @@
...
-    "start": "webpack-dev-server --mode=development --open --config developer/webpack.dev.config.js",
```
Comment (sherakama) [inline package.json:12]: start is built in to `npm run dev` now

File: core/src/templates/components/card/card.json (hunk 5 above)
Comment (yvonnetangsu) [reply core/src/templates/components/card/card.json:3]: placekitten is still alive through https://placekitten.com/ 😸

**Comment by sherakama** [comment]
From this project side, I am pretty much done, I'm currently working through the upgrade to the website so that I can get the preview build going again. https://github.com/SU-SWS/decanter-web/pull/142

**Comment by sherakama** [comment]
@yvonnetangsu @rebeccahongsf @jenbreese @jdwjdwjdw 

This is ready for your review. I have updated the steps to test. Please let me know if you need any support here. Package updates could be a pain but the rest should not have changed.

File: launch-preview.js [hunk 8]
```diff
@@ -2,21 +2,32 @@ const {execSync} = require('child_process');
...
+console.log('Current Branch:', branch);
 
 // Try deleting the styleguide if it exists. (cache)
+console.log('Starting...');
 execSync('rm -Rf ./styleguide');
 
 // Clone fresh.
-execSync('git clone https://github.com/su-sws/decanter-web.git styleguide');
+console.log('Cloning...');
+execSync('git clone -b task/package-updates https://github.com/su-sws/decanter-web.git styleguide');
```
Comment (sherakama) [inline launch-preview.js:14]: Change back to main branch before merging.

**Comment by sherakama** [comment]
Bravely going forward with this. Thank you everyone for your help.

**Comment by yvonnetangsu** [comment]
Thank you for the work @sherakama ! 🚀

**Pull request merged**